index.ts

```diff
@@ -16,7 +16,49 @@ interface MemoryHandle {
   readString(): string;
 }
 
-type Input = string | ArrayBuffer | object;
+// Provides access to data in Extism memory
+export class MemoryView {
```

Comment:
is it worth naming this `PluginOutput` to align with the Extism JS SDK? 

there could be confusion if we aren't actually using that same type from the SDK. but curious if it makes anything clearer to just name it something that's explicit about "this value is from the output of the plug-in function call"

Reply:
I went back and forth on that since it's also used as the return type for `mockInput` where that naming gets kind of confusing, but I'm open to changing it. If we do rename it, I think it might make sense to align the types even more, since `PluginOutput` extends `DataView` and provides some additional methods.

Reply:
Ah yea great points... didn't consider the mock input case. 

I'm also not certain using the exact `PluginOutput` type from the SDK works in wasm. 

since we're already separating XTP tooling from Extism a bit, I think it's totally fine to use different types here. The overlap in support & convenience is enough IMO vs. leveraging the same type.